app/src/main/java/cloudcity/GPSMonitor.java

```diff
@@ -162,6 +173,48 @@ public void stopMonitoring() {
         }
     }
 
+    public void requestCurrentLocation() {
+        boolean lacksFineLocationPermission = ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED;
+        boolean lacksCoarseLocationPermission = ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED;
+        if (lacksFineLocationPermission || lacksCoarseLocationPermission) {
+            Log.wtf(TAG, "We lack ACCESS_FINE_LOCATION or ACCESS_COARSE_LOCATION permissions!");
+            return;
+        }
+
+        Executor executor = Executors.newSingleThreadExecutor();
+        Consumer<Location> locationConsumer = new Consumer<Location>() {
+            @Override
+            public void accept(Location location) {
+                Log.d(TAG, "Current location is: "+location);
+                lastLocation = location;
+                if (location != null) {
+                    Log.d(TAG, "Current location's LAT: " + location.getLatitude() + ", LNG: " + location.getLongitude());
+                } else {
+                    Log.w(TAG, "Cannot get current location's (LAT,LNG) because location was NULL");
+                }
+            }
+        };
```

Comment:
somehow i believe the LocationManager will do all of that, we just need it to async-ly get results of a single method and then we'll let the garbage collector do what it can do.

your suggestion, while valid, might be pushing it on the over-engineering and overcomplicating side.